python_modules/dagster/dagster/_core/execution/context/compute.py

```diff
@@ -505,6 +506,15 @@ def asset_key_for_output(self, output_name: str = "result") -> AssetKey:
         else:
             return asset_output_info.key
 
+    @public
+    def output_for_asset_key(self, asset_key: AssetKey) -> str:
+        """Return the output name for the corresponding asset key."""
+        node_output_handle = self.job_def.asset_layer.node_output_handle_for_asset(asset_key)
+        if node_output_handle is None:
+            check.failed(f"Asset key '{asset_key}' has no output")
+        else:
+            return node_output_handle.output_name
+
```

Comment:
Needed to be added to support reporting metadata by asset key (as opposed to `add_output_metadata`) from the context.